### PR TITLE
Update example app

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Make the plain example app run on release mode, and also run with the latest version of React Native (0.55.3 breaks the current example for Android). The structure of the app was also simplified.

### Added
- Configured app so that no changes are required to build in release mode.

### Removed
- VectorIcons/other UI dependencies, simplified app to use single scrollview with multiple buttons.
- Deprecated navigation library

### Changed
- Updated npm dependencies to latest version available. 

-->

## Tests

Ran in release mode for Android/iOS and tested that each error appeared on the dashboard.

### Outstanding Questions

We may want to think about restricting React Native (and every dependency, for that matter) to a specific version to avoid our examples breaking without warning.

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [x] Initial review of the intended approach, not yet feature complete
  - [x] Structural review of the classes, functions, and properties modified
  - [ ] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

**Comment by fractalwrench** [comment]
Screenshot of the new app layout - clicking a button immediately triggers an exception (or other intended behaviour).

screen shot 2018-05-15 at 16 05 50

File: examples/plain/android/app/build.gradle [hunk 1]
```diff
@@ -137,11 +146,10 @@ android {
 }
 
 dependencies {
-    compile project(':react-native-vector-icons')
     compile project(':bugsnag-react-native')
     compile fileTree(dir: "libs", include: ["*.jar"])
     compile "com.android.support:appcompat-v7:23.0.1"
-    compile "com.facebook.react:react-native:+"  // From node_modules
+    compile "com.facebook.react:react-native:+" // From node_modules
```
[the remark this answers is not in the document]
Comment (kattrali) [reply examples/plain/android/app/build.gradle]: [Wow.](https://github.com/facebook/react-native/issues/19259#issuecomment-389064401)

**Comment by fractalwrench** [comment]
I've pinned the gradle dependency which is the important one - I believe all the others in `package.json` lock on either a specific or minor version.

**Comment by bengourley** [comment]
> I believe all the others in package.json lock on either a specific or minor version.

If you used yarn it will lock to specific versions, including transitive dependencies (dependencies of dependencies). I can see that the `yarn.lock` file has a big diff so I guess you did, but it would be good to confirm.

**Comment by fractalwrench** [comment]
@bengourley I've updated the PR so that a recent `yarn.lock` file is present.
